### DISCUSSION TITLE
When i use source-map, the tree shaking is doesn't work.

### DISCUSSION THREAD

**Discussion by hahaiamzd** [webpack/webpack#13741] [General]
# Bug report

**What is the current behavior?**
When i use source-map, the tree shaking is doesn't work.

**If the current behavior is a bug, please provide the steps to reproduce.**
https://webpack.js.org/guides/tree-shaking/
I follow the code in this guide, but i use source-map. After webpack, i will find the square function is in the bundle.

**What is the expected behavior?**
I expect the square function is be shaken in  the bundle.

**Other relevant information:**
webpack version:  4.46.0
Node.js version: v12.16.1
Operating System:  mac os
Additional tools: vs code

**Comment by alexander-akait** [comment]
Please provide example of code

**Reply by hahaiamzd** [reply]
I found use souce-map correlate with "eval-"  will reappear the situation. 
I will provide the example later, thx.

**Reply by hahaiamzd** [reply]
https://github.com/hahaiamzd/tree-shaking-demo
In this demo, i didn't use square function in math.js, but i can find the square function in  ./dist/bundle.js.

**Reply by alexander-akait** [reply]
Please look at https://webpack.js.org/configuration/devtool/, you should use only `production` source maps for `production`
